utils/net_utils/src/download.rs

```diff
@@ -39,18 +40,22 @@ impl<R: AsyncRead + Unpin, W: Write> AsyncRead for FetchProgress<R, W> {
     fn poll_read(
         mut self: Pin<&mut Self>,
         cx: &mut Context<'_>,
-        buf: &mut [u8],
-    ) -> Poll<Result<usize, io::Error>> {
+        buf: &mut ReadBuf<'_>,
+    ) -> Poll<std::io::Result<()>> {
+        let prev_len = buf.filled().len();
         let r = Pin::new(&mut self.inner).poll_read(cx, buf);
-        if let Poll::Ready(Ok(size)) = r {
-            self.progress_bar.add(size as u64);
+        if let Poll::Ready(Ok(())) = r {
+            self.progress_bar
+                .add((buf.filled().len() - prev_len) as u64);
```

Comment:
Why do we need this change?

Reply:
Because tokio's `AsyncRead` trait is different from the `AsyncRead` trait in `futures` (use by async-std).